common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -50,12 +52,10 @@
 
   // These are only updated by readExternal() or read()
   @Nonnull
-  private Object base;
-  private long offset;
-  private int numBytes;
+  private MemoryBlock base;
 
-  public Object getBaseObject() { return base; }
-  public long getBaseOffset() { return offset; }
+  public MemoryBlock getBaseObject() { return base; }
```

Comment:
This changes the semantics, shall we return `base.getBaseObject`?

Reply:
We can drop this function by preparing `public MemoryBlock getMemoryBlock() { return base; }`.